### PR TITLE
Add macros/functions required for azure iot sdk c

### PR DESCRIPTION
**Pull request by damianloew**
## Description

- include: add pthread_exit() missing declaration
- include: add TCP_KEEPIDLE and TCP_KEEPINTVL to tcp.h
- stdlib: add strtof()
- include: apply clang-format to math.h
- include: add isnan(), isinf() and signbit() macros to math.h

## Motivation and Context

Macros/functions that are used in azure sdk and may be useful in other incoming ports.

JIRA: PD-251

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by agkaminski — APPROVED** [review]
LGTM

File: include/netinet/tcp.h [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #ifndef _LIBPHOENIX_NETINET_TCP_H_
 #define _LIBPHOENIX_NETINET_TCP_H_
 
-#define TCP_NODELAY 1
+#define TCP_NODELAY   1
+#define TCP_KEEPIDLE  4
+#define TCP_KEEPINTVL 5
```
Comment (MaciejPurski) [inline include/netinet/tcp.h:21]: I am concerned about those values. lwip defines them differently here: 
https://github.com/phoenix-rtos/lwip/blob/383bf4f3334f88668a05120e7b503e29158eb38c/src/include/lwip/sockets.h#L283
So a call to `setsockopt()` with `TCP_KEEPIDLE` or `TCP_KEEPINTVL` would actually do a different thing. Please check it.
You should either define them to be compliant with lwip or "translate" them in our lwip sockets port.

**Review by MaciejPurski — CHANGES_REQUESTED** [review]
Please check those TCP optname values.

File: include/netinet/tcp.h (hunk 1 above)
Comment (damianloew) [reply include/netinet/tcp.h]: Yes, I realise that azure does not require some specific values, so I think they should be compatible with those defined in lwip. It's corrected now.

**Review by MaciejPurski — APPROVED** [review]
LGTM

**Pull request merged**